dashboard.h

```diff
@@ -0,0 +1,165 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef DASHBOARD_H
+#define DASHBOARD_H
+
+#include "../Bricks/port.h"
+
+#include <string>
+#include <vector>
+
+#include "../Bricks/cerealize/cerealize.h"
+#include "../Bricks/file/file.h"
+
+namespace dashboard {
+
+struct Config {
+  const std::string layout_url;
+
+  // TODO(dkorolev): Multiple production sub-domains should be listed here.
+  std::vector<std::string> data_hostnames;
+
+  // The static template.
+  std::string dashboard_template;
+
+  explicit Config(const std::string& layout_url = "/layout")
+      : layout_url(layout_url),
+        dashboard_template(
+            bricks::FileSystem::ReadFileAsString(bricks::FileSystem::JoinPath("static", "template.html"))) {}
+
+  template <typename A>
+  void save(A& ar) const {
+    ar(CEREAL_NVP(layout_url), CEREAL_NVP(data_hostnames), CEREAL_NVP(dashboard_template));
+  }
+};
+
+struct Meta {
+  struct Options {
+    std::string caption = "<CAPTION>";
+    std::string color = "blue";
+    // double min = -5;
+    // double max = 25;
+    double time_interval = 20000;
+    template <typename A>
+    void save(A& ar) const {
+      // TODO(dkorolev): Make serialization of `min/max` conditional on their equality.
+      ar(cereal::make_nvp("header_text", caption),
```

Comment:
I suggest we revisit field naming once the demo is in stable state. Chances are, we won't need to change anything.

(Not sure why did I rename this (and one other) field internally. As a hint/suggestion, I'd guess. :-) )